### DISCUSSION TITLE
Unable to restore source code

### DISCUSSION THREAD

**Discussion by arshad506** [dotnet/aspnetcore#27868] [Q&A] [answered]
I am getting below error while restoring the aspnetcore source code.

C:\src2\aspnetcore\src\Mvc>dotnet restore --interactive
  v14.15.0
C:\Program Files\dotnet\sdk\3.1.103\NuGet.targets(123,5): error : Unable to load the service index for source https://pkgs.dev.azure.com/dnceng/_packaging/darc-int-dotnet-extensions-877455ff/nuget/v3/index.json. [C:\src2\aspnetcore\src\Mvc\Mvc.sln]
C:\Program Files\dotnet\sdk\3.1.103\NuGet.targets(123,5): error :   Response status code does not indicate success: 401 (Unauthorized). [C:\src2\aspnetcore\src\Mvc\Mvc.sln]

Need help here.

**Comment by Pilchie** [comment, marked as answer]
Are you building a particular tag?  If so, edit the `NuGet.config` file to remove the `darc-int` feeds.  They are internal only feeds used for building security patches.

**Reply by arshad506** [reply]
Yes I am building version 3.1.5. After removing darc-int feeds, I am able to build now. But when I build the same solution using Visual Studio, its failing. Getting below error.

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0006	Metadata file 'C:\S6\aspnetcore\src\Components\Forms\src\bin\Debug\netstandard2.0\Microsoft.AspNetCore.Components.Forms.dll' could not be found	Microsoft.AspNetCore.Components.Web (netstandard2.0)	C:\S6\aspnetcore\src\Components\Web\src\CSC	1	Active


Can you please suggest here?

**Reply by Pilchie** [reply]
That implies that the .sln you opened in missing one of it's dependencies. Our normal builds don't use the .sln files, so they are sometimes incomplete.  You may have to either build on the command line first, or add the missing dependencies to the .sln file.

**Comment by arshad506** [comment]
The solution was build using "dotnet build" command. And now its building without any errors. Now I want to explore the source code. "app.UseStaticFiles();". When I right click on UseStaticFiles() and wanted to go to implementation, it says. "The symbol has no Implementation". Looks like the Mvc source code is using the compiled version of "Middleware". How to navigate to source code of middleware from MVC project.

Can you pls help here?.

**Comment by Pilchie** [comment]
Yes, unfortunately that's the way it works in the 3.1 branch.  The `.sln` doesn't necessarily include all of the projects in the repo. In .NET 5, we switched to a single .sln file and solution filter files for individual subsets which might help here.  In the mean time, you could add the dependent project to the solution, either by using Right Click -> Add -> Existing Project, or `dotnet sln add`.

Finally, you might consider using the online source index at https://source.dot.net/ to browse sources that aren't in your sln if you aren't planning on changing them.
